docs/sequence.rst

```diff
@@ -0,0 +1,672 @@
+
+::
+
+    Seq.fromList :: [a] -> Seq a
+    Seq.FromList xs = ...
+
+:seq:`fromList` creates a sequence containing the elements of the list
+``xs``. Sequences allow duplicate so all elements will be included in the order
+given.
+
+::
+
+    Seq.fromList ["base", "containers", "QuickCheck"]
+    > fromList ["base","containers","QuickCheck"]
+
+    Seq.fromList [0, 1, 1, 2, 3, 1]
+    > fromList [0,1,1,2,3,1]
+
+Adding to an existing sequence
+""""""""""""""""""""""""""""""
+
+::
+
+    (<|) :: a -> Seq a -> Seq a
+    x <| xs = ...
+
+    (|>) :: Seq a -> a -> Seq a
+    xs |> x = ...
+
+    (><) :: Seq a -> Seq a -> Seq a
+    l >< r = ...
+
+- ``x <| xs`` places the element ``x`` at the beginning of the sequence ``xs``..
+
+- ``xs |> x`` places the element ``x`` at the end of the sequence ``xs``.
+
+- ``l >< r`` combines the two sequences ``l`` and ``r`` together.
+
+
+Create a list from a sequence
+"""""""""""""""""""""""""""""
+
+::
+
+    foldr (:) [] :: Seq a -> [a]
+
+
+There is no ``toList`` function in the Sequence module since it can be
+`easily implemented <https://wiki.haskell.org/Foldable_and_Traversable>`_ with a
+fold using ``Seq``'s `Foldable
+<https://wiki.haskell.org/Typeclassopedia#Foldable>`_ instance.
+
+::
+
+    foldr (:) [] (Seq.fromList ["base", "containers", "QuickCheck"])
+    > ["base","containers","QuickCheck"]
+
+
+Pattern Matching
+^^^^^^^^^^^^^^^^
+
+*Since 0.5.10*
+
+Just like you can pattern match (aka. destructure) a list ``[a]``, you can do
+the same with sequneces. Lets first look at how we do this with lists::
+
+    case [1, 2, 3] of
+      [] -> "empty list"
+      (x:xs) -> "first:" ++ show x ++ " rest:" ++ show xs
+    > "first:1 rest:[2,3]"
+
+
+Let's do the same thing with sequences!
+
+::
+
+    -- Imports the patterns to match on.
+    import Data.Sequence (Seq (Empty, (:<|), (:|>)))
+
+    case Seq.fromList [1, 2, 3] of
+      Empty -> "empty sequence"
+      x :<| xs -> "first:" ++ x ++ " rest:" ++ show xs
+    > "first:1 rest:fromList [2,3]"
+
+.. NOTE:: You can't copy/paste this into GHCi because it's multiple lines.
+
+You can also take an element off the end::
+
+    -- Imports the patterns to match on.
+    import Data.Sequence (Seq (Empty, (:<|), (:|>)))
+
+    case Seq.fromList [1, 2, 3] of
+      Empty -> "empty sequence"
+      xs :|> x -> "last element:" ++ show x
+    > "last element:3"
+
+Querying
+^^^^^^^^
+
+Check if a sequence is empty
+""""""""""""""""""""""""""""
+
+::
+
+    Seq.null :: Seq a -> Bool
+    Seq.null xs = ...
+
+:seq:`null` returns ``True`` if the sequence ``xs`` is empty, ``False``
+otherwise.
+
+::
+
+    Seq.null Seq.empty
+    > True
+
+    Seq.null (Seq.fromList [1, 2, 3])
+    > False
+
+The length/size of a sequence
+"""""""""""""""""""""""""""""
+
+::
+
+    Seq.length :: Seq a -> Int
+    Seq.length xs = ...
+
+:seq:`length` returns the length of the sequence ``xs``.
+
+::
+
+    Seq.length Seq.empty
+    > 0
+
+    Seq.size (Seq.fromList [1, 2, 3])
```

Comment:
There is no such function!

Reply:
Oops, typo! I probably got it confused with the `size` in Set and Map. Speaking of which, why aren't they all called size?